### PR TITLE
NYC: terminate simulation at the beginning of iteration if config flag set

### LINKED ISSUE #3374: Stop simulation after secondary activities generation.
We need a way to obtain plans with generated secondary activities.
One of ways to do so - stop a simulation after the first iteration started.
One needs to 
1. add a boolean config entry, beam.output.writePlansAndStopSimulation (by default false)
2. dump all plans to output files.
3. stop simulation if the flag is true.

### PR DESCRIPTION
**Pull request by nikolayilyin**
This change is [Reviewable](https://reviewable.io/reviews/lbnl-ucb-sti/beam/3424)

### CODE REVIEW

**Comment by nikolayilyin** [comment]
test!

**Comment by nikolayilyin** [comment]
test!

**Review by dimaopen — COMMENTED** [review]
Is it possible instead of throwing an exception create a new utility app that loads the scenario, generates secondary activities, writes plans, and exits?
Or this would take a lot more effort?

File: src/test/scala/beam/sim/WritePlansAndStopSimulationTest.scala [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+      runBeam(
+        services,
+        scenario,
+        beamScenario,
+        beamExecutionConfig.outputDirectory,
+        plansMerged
+      )
+    }
+
+    exception.getMessage shouldBe "The simulation was stopped because beam.output.writePlansAndStopSimulation set to true."
```
Comment (JustinPihony) [inline src/test/scala/beam/sim/WritePlansAndStopSimulationTest.scala:43]: nit = I'm never a fan of a string comparison like this (brittle).

File: src/test/scala/beam/sim/WritePlansAndStopSimulationTest.scala [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+
+    val outputDir = Paths.get(beamExecutionConfig.outputDirectory).toFile
+    val outputFiles = outputDir.listFiles()
+    outputFiles.size shouldNot equal(0)
+    outputFiles.map(_.getName) should contain("generatedPlans.csv.gz")
```
Comment (JustinPihony) [inline src/test/scala/beam/sim/WritePlansAndStopSimulationTest.scala:48]: Should we cleanup files? Granted I think this is an overall problem - maybe we could create an overall cleanup in the after all tests....but that would be a new issue.....at the same time it allows later debugging and CI is cleared anyway....so maybe not worth it...just thoughts I guess...

**Review by JustinPihony — APPROVED** [review]
Some minor notes - otherwise lgtm

File: src/test/scala/beam/sim/WritePlansAndStopSimulationTest.scala (hunk 1 above)
Comment (nikolayilyin) [reply src/test/scala/beam/sim/WritePlansAndStopSimulationTest.scala:43]: Yes, generally I agree, but for this case we need to either check the error message or to throw a special type of error just for this case, which, in my opinion, is overkill.

**Comment by nikolayilyin** [comment]
> Is it possible instead of throwing an exception create a new utility app that loads the scenario, generates secondary activities, writes plans, and exits?
> Or this would take a lot more effort?

I was looking for this approach, and yes, there would be way more work.

**Pull request merged**